src/views/FlawEditView.vue

```diff
@@ -25,11 +27,21 @@ watch(() => props.id, () => {
 });
 
 function refreshFlaw() {
+  if (flawAlreadyFetched.value) {
+    flawAlreadyFetched.value = false;
+    return;
+  }
+
   errorLoadingFlaw.value = false;
   getFlaw(props.id)
       .then(theFlaw => {
         flaw.value = Object.assign({}, theFlaw);
         committedFlaw.value = Object.assign({}, theFlaw);
+        let pathID = (theFlaw.cve_id || theFlaw.uuid);
+        if (props.id !== pathID) {
```

Comment:
What edge case is this  handling?

Reply:
This condition might origin from me being inexperienced with Vue js. This condition should check whether the ID in the path should actually change or not and if not, the router pushing is not happening. How big of a difference in Vue (or probably any other frontend framework) is when you push exactly same URL on router vs not pushing at all? Does that bring some potential overhead/additional rendering/etc.?

Reply:
I think it can be left out then--  since `flaw` is a `ref` the line `flaw.value = Object.assign({}, theFlaw);` will trigger a re-render. My understanding is that any reactive values that are bound to the template will trigger a re-render, so you wouldn't have to do a reload of the page. Good consideration/question

Reply:
Thanks for clarifying off-thread that this was for replacing the URL, that makes sense--we may want to use `history.replaceState` here instead after the flaw is refreshed. That way we can remove the flag and not have to push an additional navigation

Reply:
@superbuggy thank you! Made code changes based on our discussion, please rereview.

Reply:
Hey guys, I didn't catch this. 
Why an additional navigation would be required on `FlawEditView.vue` without clearing the history state?

Reply:
@C-Valen I am not sure if I understand your question, but I've initially used the router.push to push URL with CVE ID instead of UUID and performed a logic with flag which preventeded refetching the flaw, however then @superbuggy correctly pointed out that history.replaceState should be sufficient. I originally avoided history.replaceState as I read in the Vue.js docs that not all globals (like window, history, etc.) are not enabled by default and got false impression that they should be avoided, however the this apparently holds only for templates on not the script part, in the script part they are available. Hopefully this explains the problem.

Reply:
Thanks @JakubFrejlach. So the purpose of `history.replaceState` is to prevent refetching the flaw?

Reply:
Ok, makes sense to me @JakubFrejlach. I was not seeing that scenario where an user could access directly by url to /flaws/<uuid> in which case the url indeed should be replaced to use the CVE ID (if available).
Thank you for the clarification!

Reply:
Thanks!